babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -1586,46 +1586,27 @@ SqlNode SqlSelectTopN(SqlParserPos pos) :
 
 SqlNode SqlHexCharStringLiteral() :
 {
-    final NlsString p;
+    final String hex;
     final String formatString;
-    final String charSet;
-    CharacterSet charSetEnum = null;
-    final HexCharLiteralFormat format;
+    String charSet = null;
 }
 {
     (
         <PREFIXED_HEX_STRING_LITERAL>
         {
-            charSet = SqlParserUtil
-                      .trim(SqlParserUtil.getCharacterSet(token.image), " ")
-                      .toUpperCase();
-            if (charSet.equals("LATIN")) {
-                charSetEnum = CharacterSet.LATIN;
-            } else if (charSet.equals("UNICODE")) {
-                charSetEnum = CharacterSet.UNICODE;
-            } else if (charSet.equals("GRAPHIC")) {
-               charSetEnum = CharacterSet.GRAPHIC;
-            } else if (charSet.equals("KANJISJIS")) {
-              charSetEnum = CharacterSet.KANJISJIS;
-            } else {
-              throw SqlUtil.newContextException(getPos(),
-              RESOURCE.unknownCharacterSet(charSet));
-            }
+            charSet = SqlParserUtil.getCharacterSet(token.image);
         }
     |
         <QUOTED_HEX_STRING>
     )
     {
+        // In the case of matching "PREFIXED_HEX_STRING_LITERAL" or
+        // "QUOTED_HEX_STRING" token, it is guaranteed that the following
+        // Java string manipulation logic is valid.
         String[] tokens = token.image.split("'");
-        p = new NlsString(tokens[1], null, null);
+        hex = tokens[1];
         formatString = tokens[2];
-        if (formatString.equals("XC")) {
-            format = HexCharLiteralFormat.XC;
-        } else if (formatString.equals("XCV")) {
-            format = HexCharLiteralFormat.XCV;
-        } else {
-            format = HexCharLiteralFormat.XCF;
-        }
-        return new SqlHexCharStringLiteral(p, getPos(), charSetEnum, format);
+        return new SqlHexCharStringLiteral(hex, getPos(), charSet,
+        formatString);
```

Comment:
nit: Looks like this needs more indentation.

Reply:
Fixed in https://github.com/googleinterns/calcite/pull/155/commits/5cb9a502550b70ab3bc5a70a741cb4b0c645e14d